common/lib/client/auth.js

```diff
@@ -97,7 +108,33 @@ var Auth = (function() {
 			Logger.logAction(Logger.LOG_ERROR, 'Auth()', msg);
 			throw new Error(msg);
 		}
-	}
+	};
+
+	/**
+	 * Update the authentication options for this client;
+	 * this will also update these options on the realtime connection
+	 * so that they take effect immediately
+	 * @param options
+	 */
+	Auth.prototype.setOptions = function(options) {
```

Comment:
> If you wanted to set a new token with new capabilities, how would authorise() do that?

In a number of ways:

```
/* If generating the token */
client.auth.authorise({ capability: newCapability }, { force: true })

/* When using an external authUrl */
client.auth.authorise(null, { force: true })
```

`setOptions` is very misleading.  Is it setting options for future requests, is it setting options for the client library?  Why is setOptions communicating to the server, and more importantly, why is it causing the client to disconnect.

`authorise` is clear, it's telling the client to authorise, and with `force: true` it's stating force a reauth.